### ISSUE TITLE
Searching: extension to stop filtering

### ISSUE THREAD

**Issue by lecc0r** [JabRef/jabref#444]
The newly implemented search function is great and helps a lot to search/filter my 3000 paper data base. However, after having done my search I want to see again all entries. Therefore, I have to stop the searching/filtering by either clicking into the search field or press Ctrl + F and delete the search term. 
I would like to suggest a handy extension to stop the searching/filtering: either a stop button next to the search field or alternatively stop the searching/filtering by hitting the Escape button on the keyboard. Both would be preferable.

**Comment by simonharrer** [comment]
Will implement this for the Escape button.

**Comment by tobiasdiez** [comment]
What about displaying a delete/clear icon in the search field, as for example here:
![Image](https://wiki.sugarlabs.org/images/7/7f/Example_searchfield_suggestions.png)

**Comment by matthiasgeiger** [comment]
@tobiasdiez Yes, that would be the perfect solution. Do you want to implement it? 
Potential starting point might be this blogpost: https://tips4java.wordpress.com/2009/09/27/component-border/ 
A fitting icon would be https://materialdesignicons.com/icon/close-circle

**Comment by simonharrer** [comment]
Good idea, but I do not want to have too much swing UI code before changing at some point in time to JavaFX. Hence, I aim for simple UI solutions. But as a user I would like to have such a feature.

**Comment by matthiasgeiger** [comment]
The simpler, and not so fancy solution would be a simple button next to the search field ;-)

**Comment by stefan-kolb** [comment]
ESC binding is still appropriate as additional option, like in Intellij.

**Comment by tobiasdiez** [comment]
For the moment, I second the idea of a button next to the search field. I will gladly implement the inline icon after we switched to JavaFX.

**Comment by simonharrer** [comment]
- what should the button say: "Clear", "Reset", "Cancel", display an X?
- where should the ESC behavior apply: only in the text field? when the focus is also in the main table?

**Comment by lecc0r** [comment]
* I would go for "Clear"
* Ideally it should apply to both. When searching for an entry I often get more than one result and I then have to click on the one I was searching for. Thus, the focus is often in the main table. But it should always be able to quickly clear the search field by hitting ESC when the focus is in the text field.

**Comment by simonharrer** [comment]
At the moment, ESC is already bound on the main table. You can just enter letters and numbers when the main table is selected and it automatically should find the entry that starts with the entered letters and numbers. ESC ends this jumping around. 

Hence, I would propose to only implement the button + ESC with focus in search field.

**Comment by simonharrer** [comment]
Please have a look at the current master build. It contains the implemented feature.

**Comment by lecc0r** [comment]
@simonharrer Yes, seems to work fine!

**Comment by koppor** [comment]
While working with current master, I found out that the current solution is disturbing for my workflow. I would have expected that the first "ESC" closes the autocomplete dropdown. The second "ESC" focues the main table and the third "ESC" clears the search.

I used to search for something (in floating mode), then press ESC and then use the cursor up- and down keys to go through the found entries (because I read the entry preview). Now, this is not possible. How can I achieve my old behavior with the current solution? Would the mulitple-ESC-thingy a way to go for the others, too?
